### PR TITLE
Add mixfile snippet on package page

### LINKED ISSUE #110: Provide copy/pastable snippet on package page for mix.exs entry
It could be for the latest version, i.e.: If I'm viewing httpoison's hex.pm page, I could have a snippet:

    {:httpoison, "~> 0.6"}

That is for the latest major version. Thoughts?

### PR DESCRIPTION
**Pull request by jcspencer**
Closes #110

:tomato:

### CODE REVIEW

File: lib/hex_web/web/router.ex [hunk 1]
```diff
@@ -162,6 +163,15 @@ defmodule HexWeb.Web.Router do
     end)
   end
 
+  defp snippet_version({:ok, version}) do
```
Comment (ericmj) [inline lib/hex_web/web/router.ex]: There should be no pattern matching here.

File: lib/hex_web/web/templates/package.html.eex [hunk 2]
```diff
@@ -3,6 +3,14 @@
     <a href="/packages/<%= @package.name %>"><%= @package.name %></a>
     <%= if @current_release do %>
       <small><%= @current_release.version %></small>
+
+      <span class="form-inline pull-right">
+        <input type="text" class="form-control input-sm snippet"
+          value='{:<%= @package.name %>, "<%= @snippet %>"}'
+          onfocus="this.select();"
+          onmouseup="return false;"
```
Comment (ericmj) [inline lib/hex_web/web/templates/package.html.eex]: What does the onmouseup do?

File: priv/static/static/main.css [hunk 3]
```diff
@@ -111,3 +111,8 @@ body {
   padding-right: 1.5em;
   text-align: right;
 }
+
+.snippet {
+  cursor: default;
+  background: transparent;
```
Comment (ericmj) [inline priv/static/static/main.css]: Why do we need this background?

Comment (jcspencer) [inline priv/static/static/main.css]: When using `readonly`, the input gets a nasty, gray background. Setting the background to `transparent` sets it back to normal.

File: lib/hex_web/web/router.ex (hunk 1 above)
Comment (jcspencer) [inline lib/hex_web/web/router.ex]: What do you suggest as an alternative, seeing as you wanted to have the function accept the output of `Version.parse`?

File: lib/hex_web/web/templates/package.html.eex (hunk 2 above)
Comment (jcspencer) [inline lib/hex_web/web/templates/package.html.eex]: Whoops, I didn't realize I'd left that there; it's not used.

**Comment by josevalim** [comment]
So should we really do this? We want to be the package manager for the Erlang VM. Does it mean we are going to show Rebar snippets too? Or whatever comes after?

**Comment by josevalim** [comment]
Also, I would put less important on the build tool in the package page. From a first glance, it is the thing that calls the most attention but it is definitely not the most important. Maybe we could unify the build tool with the part that shows the version. So, if we add rebar, they would be able to click rebar and see rebar instructions.

**Comment by josevalim** [comment]
Also, when we add escripts, there may be packages that are meant to be used as escripts and not to be installed on mix, which could be a bit confusing if we show the mix.exs information.

All I am saying is, we need to better think how we are going to handle all those cases. For example, rubygems shows both `gem install` and Gemfile instructions.

File: lib/hex_web/web/router.ex (hunk 1 above)
Comment (ericmj) [inline lib/hex_web/web/router.ex]: `Version.parse` doesn't seem to be called though.

**Comment by ericmj** [comment]
I talked to @josevalim and we decided to hide the build tools behind some expandable box with more explanation because it draws too much attention today and doesn't make sense without more explanation, for example cowboy shows "[make] [rebar]" but it actually compiles with mix. This is not the scope of this issue though.

Like @josevalim said we need to display the dependency definition for both mix and rebar since this package manager is not for mix only. We need to move it down under the "Downloads" box or as a third column beside "Versions" and "Dependencies". We can have each definition behind a tab or do like GitHub does for clone URL where you can click to change between protocols [1]. Or do just like rubygems that displays both [2].

[1] 
screen shot 2015-05-09 at 20 19 48

[2] 
screen shot 2015-05-09 at 20 22 32

**Comment by ericmj** [comment]
Are you still working on this James?

**Comment by jcspencer** [comment]
I'm happy to go ahead, but I'm not sure which snippets would be most appropriate. Would the basics simply be Rebar and Mixfile snippets? Do we currently have a way to determine if a package should not be added as a snippet (e.g. escripts)?

**Comment by ericmj** [comment]
Lets go with rebar and mix. We can add escripts later when we add explicit support for it.

**Comment by jcspencer** [comment]
So practically `{cowboy, "1.0.0"}` for Rebar and `{:cowboy, "~> 1.0.0"}` for Mix?

**Comment by ericmj** [comment]
@jcspencer Yes

**Comment by ericmj** [comment]
Ping?

**Comment by ericmj** [comment]
Any updates on this @jcspencer? Anything I can help with?

**Comment by ericmj** [comment]
If you have stopped working on this @jcspencer please let us know so someone else can grab it.

**Comment by jcspencer** [comment]
@ericmj I think I mentioned this on IRC; I somehow `rm -rf`ed my git tree, so I will have to start from scratch on this one. If someone else can grab it, that would be great.

**Pull request closed without merging**